junk-drawer/cache.janet

```diff
@@ -0,0 +1,18 @@
+(defn cache-insert [self query result]
+  (put self query result)
+  result)
+
+(defn cache-get [self query]
+  (get self query))
+
+(defn cache-partial-clear [self component]
+  (loop [query :keys self
+         :when (index-of component query false)]
+    (put self query nil)))
+
+(defn init []
+  (table/setproto
+   @{}
+   @{:insert cache-insert
```

Comment:
I would remove the `cache-` prefix from the function names.

Reply:
Yaaa, I coded myself into a corner by naming one of the functions `get`, and I didn't want to override the standard API. The functions are small enough that I could probably get away with in-lining them right into the prototype object, but that kind of feels wrong also.

Reply:
That makes sense. If functions will not be part of the public API I am using good old _ at the start of the name in these cases, together with `defn-`. Which indeed does not feel great too

Reply:
ah ya totally forgot to use `defn-`, `_get` is good to me!